generalov/nodejs-aws-be
# task9

# Task 9 (Backend For Frontend)
https://github.com/rolling-scopes-school/nodejs-aws-tasks/blob/main/task9-bff/task.md

---
Frontend: https://dibzgqdmk7fo5.cloudfront.net/ (PR: https://github.com/generalov/nodejs-aws-fe/pull/7)

The cache should expire in 2 minutes.  
How to test (using frontend):
  * Get products list (on the home page)
  * Create new product
  * Get products list - result shouldn’t have new product
  * Wait more than 2 minutes
  * Get products list - result should have new product

You can add a query param to the product service BFF URL. This requests fresh data from the product API (and then puts another item into the cache):
https://24jnm3v457.execute-api.us-east-1.amazonaws.com/dev/product
https://24jnm3v457.execute-api.us-east-1.amazonaws.com/dev/product?foo

Cart API is not cached:
Cart BFF: https://24jnm3v457.execute-api.us-east-1.amazonaws.com/dev/cart/profile/cart

---

Product BFF API Gateway: https://24jnm3v457.execute-api.us-east-1.amazonaws.com/dev/product
Product BFF: http://generalov-bff-api-devel.us-east-1.elasticbeanstalk.com/product
Product API: https://mzvml8midc.execute-api.us-east-2.amazonaws.com/dev/product

Cart BFF API Gateway: https://24jnm3v457.execute-api.us-east-1.amazonaws.com/dev/cart
Cart BFF: http://generalov-bff-api-devel.us-east-1.elasticbeanstalk.com/cart
Cart API: http://generalov-cart-api-devel.us-east-2.elasticbeanstalk.com/api

`yarn add  --dev @nestjs/cli -W`
`yarn nest new --directory services/bff-service --skip-git --skip-install bff-service`

## Evaluation criteria (each mark includes previous mark criteria)

 
## Additional (optional) tasks
- [x] * **+1** - Add a cache at the **bff-service** level for a request to the **getProductsList** function of the **product-service**.
